nifi-nar-bundles/nifi-jwt-bundle/nifi-jwt-processors/src/main/java/org/apache/nifi/processors/jwt/UnsignJWT.java

```diff
@@ -0,0 +1,236 @@
+import org.apache.nifi.processor.io.OutputStreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.bouncycastle.util.encoders.Base64;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.List;
+import java.util.Set;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map.Entry;
+
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@SideEffectFree
+@Tags({ "jwt", "json", "flowfile", "unsign" })
+@CapabilityDescription("Validates and unsigns a JWT either from the flowfile body or from an attribute. The JWT is unsigned "
+                + "using a directory of public RSA keys specified by the \"Approved Public Keys\" property. By default the "
+                + "processor will read the JWT from the content of the flowfile and rewrite the contents with the JWT claims "
+                + "in JSON format. If an attribute name is given as the \"JWT Attribute Name\" property then the content of "
+                + "that attribute will be unsigned and the JWT claims will be written as \"jwt.XXX\" attributes.")
+@WritesAttributes({
+        @WritesAttribute(attribute = "jwt.header", description = "If the flow file is routed to the success relationship "
+                + "the attribute will contain the the header of the JWT."),
+        @WritesAttribute(attribute = "jwt.footer", description = "If the flow file is routed to the success relationship "
+                + "the attribute will contain the footer of the JWT."),
+            @WritesAttribute(attribute = "jwt.XXX", description = "If the flow file is routed to the success relationship "
+                + "and the JWT was unsugned from an attribute, these attributes will contain the claims of the JWT."),
+        @WritesAttribute(attribute = "failure.reason", description = "If the flow file is routed to the failure relationship "
+                + "the attribute will contain the error message resulting from the validation failure.") })
+@SeeAlso({ SignJWT.class })
+public class UnsignJWT extends AbstractProcessor {
+
+    public static final PropertyDescriptor PUBLIC_KEYS_PATH = new PropertyDescriptor.Builder().name("PUBLIC_KEYS_PATH")
+            .displayName("Approved Public Keys")
+            .description("Path to the directory containing approved public certificates. Certificates are expected to have "
+                + "extension \".pub\" and be x509 PEM files")
+            .required(true).addValidator(CustomValidators.DIRECTORY_HAS_PUBLIC_KEYS_VALIDATOR).build();
+
+    public static final PropertyDescriptor JWT_ATTRIBUTE_NAME = new PropertyDescriptor.Builder().name("JWT_ATTRIBUTE_NAME")
```

Comment:
Is there a reason we don't want to support Expression Language in this property?

Reply:
No reason I can think of, will add EL support :)